components/omega/src/ocn/VertMix.h

```diff
@@ -0,0 +1,183 @@
+                                                       DelTangVel * DelTangVel);
+            }
+            Real DelZMid = ZMid(ICell, K - 1) - ZMid(ICell, K);
+            ShearSquared = ShearSquared / (DelZMid * DelZMid);
+
+            Real RichardsonNum = BruntVaisalaFreq(ICell, K) /
+                                 Kokkos::max(1.0e-12_Real, ShearSquared);
+
+            ShearViscVal =
+                ShearNuZero / Kokkos::pow(1.0_Real + ShearAlpha * RichardsonNum,
+                                          ShearExponent);
+            VertVisc(ICell, K) = VertVisc(ICell, K) + ShearViscVal;
+            VertDiff(ICell, K) =
+                VertDiff(ICell, K) +
+                ShearViscVal / (1.0_Real + ShearAlpha * RichardsonNum);
+         }
+      }
+   }
+
+ private:
+   Array1DReal DcEdge;
+   Array1DReal DvEdge;
+   Array1DReal AreaCell;
+   Array2DReal ZMid;
+   Array1DI4 NEdgesOnCell;
+   Array2DI4 EdgesOnCell;
+   I4 NVertLayers;
+};
+
+/// Class for Vertical Mixing Coefficient (VertMix) calculations
+class VertMix {
+ public:
+   /// Get instance of VertMix
+   static VertMix *getInstance();
+
+   /// Destroy instance (frees Kokkos views)
+   static void destroyInstance();
+
+   Array2DReal VertDiff; ///< Vertical diffusivity field (m^2 s^-1)
+   Array2DReal VertVisc; ///< Vertical viscosity field (m^2 s^-1)
+
+   std::string VertDiffFldName;  ///< Field name for vertical diffusivity
+   std::string VertViscFldName;  ///< Field name for vertical viscosity
+   std::string VertMixGroupName; ///< VertMix group name (for config)
+   std::string Name;             ///< Name of this VertMix instance
+
+   // Background mixing parameters
+   Real BackDiff = 1.0e-5; ///< Background vertical diffusivity (m^2 s^-1)
+   Real BackVisc = 1.0e-4; ///< Background vertical viscosity (m^2 s^-1)
+
+   ConvectiveMix
+       ComputeVertMixConv;       ///< Functor for Convective VertMix calculation
+   ShearMix ComputeVertMixShear; ///< Functor for Shear VertMix calculation
+
+   /// Compute vertical diffusivity and viscosity for all cells/layers
+   void computeVertMix(const Array2DReal &NormalVelocity,
+                       const Array2DReal &TangentialVelocity,
+                       const Array2DReal &BruntVaisalaFreq);
+
+   /// Initialize VertMix from config and mesh
+   static void init();
+
+ private:
+   /// Private constructor
+   VertMix(const std::string &Name, const HorzMesh *Mesh,
+           const VertCoord *VCoord);
+
+   /// Private destructor
+   ~VertMix();
+
+   static VertMix *Instance; ///< Instance pointer
+
+   // Delete copy and move constructors and assignment operators
+   VertMix(const VertMix &)            = delete;
+   VertMix &operator=(const VertMix &) = delete;
+   VertMix(VertMix &&)                 = delete;
+   VertMix &operator=(VertMix &&)      = delete;
+
+   I4 NCellsAll;   ///< Number of horizontal cells
+   I4 NEdgesAll;   ///< Number of horizontal edges
```

Comment:
This member is never used. In fact, it is never even initialized. The same is true for `NVertLayers`, but that variable may be needed if this class is changed to do parallel computations in the vertical.